scripts/globals/events/starlight_celebrations.lua

```diff
@@ -0,0 +1,943 @@
+-----------------------------------
+-- Starlight Celebrations
+-----------------------------------
+require("scripts/globals/settings")
+require("scripts/globals/status")
+require("scripts/globals/utils")
+require("scripts/globals/npc_util")
+------------------------------------
+
+xi = xi or {}
+xi.events = xi.events or {}
+xi.events.starlightCelebration = xi.events.starlightCelebration or {}
+
+function xi.events.starlightCelebration.isStarlightEnabled()
+    local option = 0
+    local month = tonumber(os.date("%m"))
+    local day = tonumber(os.date("%d"))
+
+    if ((month == 12 and day >= 16) or xi.settings.main.STARLIGHT_YEAR_ROUND) then -- According to wiki Startlight Festival is December 16 - December 31.
```

Comment:
I'm assuming you'll get it from someone, so I'll be the first to say it.
nit: don't use optional parenthesis in your ifs, so this can just be:
```lua
if (month == 12 and day >= 16) or xi.settings.main.STARLIGHT_YEAR_ROUND then -- According to wiki Startlight Festival is December 16 - December 31.
```

Reply:
Will do!

Reply:
All set, merging this change shortly!